### ISSUE TITLE
w32- functions are undefined

### ISSUE THREAD

**Issue by jabranham** [emacs-ess/ESS#518]
ESS has many calls to functions starting with `w32-`. These aren't defined for me (GNU Emacs 26.1 RC1) and produce errors when, for example, trying to open a BUGS/JAGS file:

```
(void-function w32-shell-dos-semantics)
  w32-shell-dos-semantics()
  ess-jags-mode()
```

Does anyone know what these functions are supposed to do? I can't find any mention of them online or in Emacs's documentation/NEWS files. Can we remove them?

**Comment by ess-core** [comment]
You must keep them.  They are necessary on emacs running on Windows.
Perhaps you need to condition on something rather than run them unconditionally.

On Macintosh
GNU Emacs 25.1.1 (x86_64-apple-darwin13.4.0, NS appkit-1265.21 Version
10.9.5 (Build 13F1911)) of 2016-09-20
ess-version: 16.10-1 [<unknown>] (loaded from
/Applications/Emacs.app/Contents/Resources/site-lisp/ess/)

M-x w32-<tab>
reports [no match]

On the Macintosh,  w32-shell-dos-semantics is defined in
ess/lisp/ess-compat.el



On Windows
GNU Emacs 25.1.1 (i686-w64-mingw32) of 2016-11-15
ess-version: 16.10 [Released git: e92ed78b8890] (loaded from
c:/Program Files (x86)/GNU Emacs 25.1/share/emacs/site-lisp/)

M-x w32-<tab>
gives 12 options for completion.

The one you mention
 w32-shell-dos-semantics()
isn't in the list of completions because it does not include the line
(interactive).
C-h f w32-shell-dos-semantics
is defined in file  ‘emacs/25.1/lisp/w32-fns.el’.

**Comment by jabranham** [comment]
Thanks. Surely they're only necessary for windows on 32-bit systems? I'd hope that not too many people are using that anymore.... 

It seems like they aren't defined on GNU/Linux systems because w32-fns isn't loaded (and if it's tried, I get an error). 

I've wrapped them in calls to `fboundp` so that they don't produce warnings when byte compiled and error when tried to run in #515
